api/src/main/java/pl/kolendateam/dadcard/feats/entity/CharacterFeat.java

```diff
@@ -30,4 +36,95 @@ public int findFeatIndexinArrayById(ArrayList<CharacterFeat> featPcList){
     public void incrementLevelFeat(){
         this.levelOfFeat++;
     }
+
+    public int checkPrerequisite(
+        Feats feat,
+        String subRace,
+        SavingThrow savingThrow, ArmorClass armorClass,
+        ArrayList<ClassSkills> classSkills, Abilitys abilitys,
+        int bab,
+        ArrayList<CharacterFeat> featsList) {
+
+        Prerequisite p = new Gson().fromJson(feat.getPrerequisite(), Prerequisite.class);
+
+        boolean checkPrerequisite = false;
+
+        if(p.getRace()!=null){
+            for(String r : p.getRace()){
+                if(subRace == r){
+                    checkPrerequisite = true;
+                }
+            }
+        }
+
+        if(p.getSavingThrow()!=null){
+            if(savingThrow.checkPrerequisiteST(p.getSavingThrow())==1){
+                checkPrerequisite = true;
+            } else {
+                checkPrerequisite = false;
+            }
+        }
+
+        if(p.getArmorClass()!=null){
```

Comment:
te if sa troche bez sensu bo ostatni napisze ci checkPrerequisite na true nawet jeszt wszystko wczesniej dawalo false

Reply:
checkPrerequisite powinien byc poczatkowo true i tylko jak coś jest nie tak ustawiac sie na false

Reply:
racja, zmienie to by policzył ile razy nie ma nulla w prerekwisycie i potwierdzi czy mozna dodać czy nie